### DISCUSSION TITLE
GitHub Apps and user accounts

### DISCUSSION THREAD

**Discussion by quinnlovescode** [community/community#24911] [Apps, API and Webhooks]
<p>Is it possible to build a GitHub App that takes actions on behalf of my user account?</p>

**Comment by nadiajoyce** [comment, marked as answer]
<p>Hi <a class="mention" href="/quinnlovescode">@quinnlovescode</a>!</p>
<p>Technically, yes, you can do this, but it’s not the typical use case. When GitHub Apps take action through GitHub, they act as <a href="https://developer.github.com/v4/reference/object/bot/" rel="nofollow noopener">bot users</a>. By contrast, OAuth apps take action as user accounts, known as machine accounts. (Check out the differences between machine and bot accounts <a href="https://developer.github.com/apps/differences-between-apps/#machine-vs-bot-accounts" rel="nofollow noopener">here</a>.)</p>
<p>It <em>is</em> possible for a GitHub App to act on behalf of a user using a user access token, with very limited access. We call these user-to-server requests. You can find the list of supported endpoints <a href="https://developer.github.com/apps/building-github-apps/identifying-and-authorizing-users-for-github-apps/#user-to-server-requests" rel="nofollow noopener">here</a>.</p>
<p>Cheers!</p>
